### ISSUE TITLE
Planned swaybar tray features

### ISSUE THREAD

**Issue by ianyfan** [swaywm/sway#3799]
[ ]  XEmbed proxy
[ ] Canonical DBus menu (currently blocked by layer shell popups)
[ ] caching icons correctly: https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html#implementation_notes (ref https://github.com/swaywm/sway/issues/3789)
[ ] Locales when parsing index.theme file

Currently not being worked on (by me, at least) so contributions would be very welcome :)

**Comment by artizirk** [comment]
KDE guys have a XEmbed proxy daemon https://github.com/KDE/plasma-workspace/tree/master/xembed-sni-proxy

**Comment by m-khvoinitsky** [comment]
Consider adding: SVG icons support.
Currently, for example, Quassel shows sad face instead of icon because its icon is SVG.

**Comment by ianyfan** [comment]
Hmm, I did allow it to load SVGs if it was built with gdk-pixbuf2 support, but I never actually tested it, so it might have gone wrong somewhere.

**Comment by m-khvoinitsky** [comment]
I'm not 100% sure that the reason is SVG but that's the only difference that I've noticed between Quassel and working icons. And there is no any hint in logs.

**Comment by emersion** [comment]
>However, it is currently blocked by layer shell popups

This can now be implemented, as layer-shell popups are supported by Sway.

**Comment by agraven** [comment]
Should an issue be created for the implementing the DBus menu specification?

**Comment by m-khvoinitsky** [comment]
> Consider adding: SVG icons support.
Currently, for example, Quassel shows sad face instead of icon because its icon is SVG.

It works since 1.4.

**Comment by emersion** [comment]
PR implementing D-Bus menus: https://github.com/swaywm/sway/pull/5161

**Comment by nmschulte** [comment]
I'm not sure what is spec'd, but is it possible to have tray icon tooltips, that helps me identify the tray item, some sort of name or title?

**Comment by ianyfan** [comment]
Hm, I guess you can do it now that popups have been implemented, and there is a tooltip property in the SNI spec, but it's not something that I'm planning on doing, so someone else will have to implement it

**Comment by Merith-TK** [comment]
how do i enable this? running archlinux on sway 1.4. but the only tray icon i have ever seen on my system through swaybar was albert, which didnt work due to the whole isolation that wayland does

**Comment by LambLeeg** [comment]
Recently I began to use Sway instead of i3 and I only have one bug, my system tray has no menu bar with actions, for example, Steam or Telegram also i3 by default has a WiFi settings icon that I have never seen in SwayWM. Good luck with this project because Wayland is the future. Anyway, Sway has more features and I absolutely like it!

**Comment by crocket** [comment]
Both xembed-sni-proxy and snixembed failed to work on sway-1.5 on gentoo linux.

**Comment by GameWizard5000** [comment]
I was able to compile xembed-sni-proxy with this [patch](https://aur.archlinux.org/cgit/aur.git/plain/cmake.patch?h=xembed-sni-proxy-git) and with the cmake options below, but whenever a window that uses Xembed like Discord launches there's a black box on screen 
```
cmake . \
    -DCMAKE_BUILD_TYPE=Release \
    -DCMAKE_INSTALL_PREFIX=/usr \
        -DKDE_INSTALL_LIBDIR=lib \
        -DKDE_INSTALL_LIBEXECDIR=lib \
        -DBUILD_TESTING=OFF
```

**Comment by ianyfan** [comment]
Can we please stop reporting random bugs here please? If xembed proxy doesn't work, and it's _identifiably a problem with sway_, then please open a new issue. The tray was written as a standards-compliant implementation, and not to specifically work with an xembed proxy, so the places where they work are more of a happy accident. There are also no current plans to roll our own proxy either.
Please encourage xembed apps to use SNIs, thanks.

**Comment by xi** [comment]
For anyone searching for a short term solution for menues: It is not that hard to show dbus menues in dmenu or something like that. Proof of concept: https://gist.github.com/xi/b88af997b485191fa0beb5d6612ead29

**Comment by oniGino** [comment]
I can understand not support Xembed, but for SNI is there a ticket somewhere tracking dbusmenu work?

**Comment by Hikandy** [comment]
Any work-arounds for this meanwhile?

**Comment by oniGino** [comment]
> Any work-arounds for this meanwhile?

I switched to waybar, it is a lot more feature rich and has full tray support

**Comment by akhiljalagam** [comment]
> Any work-arounds for this meanwhile?

https://gist.github.com/xi/b88af997b485191fa0beb5d6612ead29

**Comment by tatsumoto-ren** [comment]
I see there's an open PR here: https://github.com/swaywm/sway/pull/8920
It would be awesome if someone reviewed it.
